lerobot/common/policies/act/modeling_act.py

```diff
@@ -128,6 +129,7 @@ def forward(self, batch: dict[str, Tensor]) -> dict[str, Tensor]:
         """Run the batch through the model and compute the loss for training or validation."""
         batch = self.normalize_inputs(batch)
         if len(self.expected_image_keys) > 0:
+            batch = dict(batch)  # shallow copy so that adding a key doesn't modify the original
```

Comment:
FYI, although this shallow copy already happens in `self.normalize_inputs` I think it's safer to add it here too rather than relying on an implicit guarantee.

Reply:
Pretty sure since it's a shallow copy the tensors are not touched. I'm not sure about torch.compile as I haven't used it, but I think we'll probably have a lot to look at regarding that.

Reply:
sounds good